### PR TITLE
MULE-11432: Disallow operations to receive the current Event and Message

### PR DESCRIPTION
**Pull request by estebanwasinger**

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OperationParametersTypeModelValidator.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  @Override
+  public void validate(ExtensionModel extensionModel, ProblemsReporter problemsReporter) {
+    new IdempotentExtensionWalker() {
+
+      @Override
+      protected void onOperation(OperationModel model) {
+        model.getAllParameterModels().forEach(parameterModel -> {
+          if (parameterModel.getType().getMetadataFormat().equals(MetadataFormat.JAVA)) {
+            Class<?> type = getType(parameterModel.getType());
```
Comment (alegmarra) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OperationParametersTypeModelValidator.java]: I'd avoid getting the Class, instead load the Event and Message MetadataTypes and check if they are equals to the parameter type

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/TypeSafeStaticValueResolver.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import org.mule.runtime.core.api.MuleContext;
+
+/**
+ * {@link ValueResolver} implementation which always returns the same value, but if it is required,
+ * the value will be converted to the required type using the mule transformers.
+ *
+ * @param <T> Type of the value to resolver
+ * @since 4.0
+ */
+public class TypeSafeStaticValueResolver<T> implements ValueResolver<T> {
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/TypeSafeStaticValueResolver.java]: This approach seems overkill.. What is the advantage of having a `TypeSafeStaticValueResolver`, plus a `TypeSafeExpressionValueResolver` which both compose a `TypeSafeTransformer` (which is not actually a transformer) vs simply having a `TypeSafeValueResolverDecorator` which can  work with any kind of ValueResolver we may need.

File: extensions/file/src/main/java/org/mule/extension/file/internal/FileOperations.java [hunk 3]
```diff
@@ -150,20 +148,18 @@
    * @param createParentDirectories whether or not to attempt creating any parent directories which don't exists.
    * @param lock                    whether or not to lock the file. Defaults to false
    * @param mode                    a {@link FileWriteMode}. Defaults to {@code OVERWRITE}
-   * @param event                   The current {@link Event}
    * @throws IllegalArgumentException if an illegal combination of arguments is supplied
    */
   @Summary("Writes the given \"Content\" in the file pointed by \"Path\"")
   @Throws(FileWriteErrorTypeProvider.class)
   public void write(@UseConfig FileConnectorConfig config, @Connection FileSystem fileSystem, @Optional String path,
```
Comment (marianogonzalez) [inline extensions/file/src/main/java/org/mule/extension/file/internal/FileOperations.java]: you need to change the javadoc (and also the spec, but that can wait) ;)

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OperationParametersTypeModelValidator.java (hunk 1 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/validation/OperationParametersTypeModelValidator.java]: there's no problem in getting the class if you already know that the format is JAVA.

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/TypeSafeExpressionValueResolver.java [hunk 4]
```diff
@@ -33,55 +27,17 @@
  */
 public class TypeSafeExpressionValueResolver<T> implements ValueResolver<T> {
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/TypeSafeExpressionValueResolver.java]: why does this class still exists?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ExpressionTypedValueValueResolver.java [hunk 5]
```diff
@@ -23,13 +22,15 @@
  * @param <T>
  * @since 4.0
  */
-public class ExpressionTypedValueValueResolver<T> extends TypeSafeExpressionValueResolver<TypedValue<T>> {
+public class ExpressionTypedValueValueResolver<T> extends ExpressionValueResolver<TypedValue<T>> {
```
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ExpressionTypedValueValueResolver.java]: why does this class still exists?

**Pull request merged**